enterprise/backend/src/metabase_enterprise/data_editing/api.clj

```diff
@@ -16,28 +23,93 @@
                            {:database (api/check-404 (t2/select-one-fn :db_id [:model/Table :db_id] table-id))
                             :table-id table-id
                             :arg      rows}
-                           :policy :data-editing))
+                           :policy   :data-editing))
+
+(doseq [topic [:event/data-editing-row-create
+               :event/data-editing-row-update
+               :event/data-editing-row-delete]]
+  (defmethod events.notification/notification-filter-for-topic topic
+    [_topic event-info]
+    (assert (:table_id event-info) "Event info must contain :table_id")
+    [:= :table_id (:table_id event-info)]))
+
+(defn- qp-result->row-map
+  [{:keys [rows cols]}]
+  ;; rows from the request are keywordized
+  (let [col-names (map (comp keyword :name) cols)]
+    (map #(zipmap col-names %) rows)))
+
+(defn- table-id->pk
+  [table-id]
+  ;; TODO: support composite PKs
+  (api/check-404 (t2/select-one :model/Field :table_id table-id :semantic_type :type/PK)))
```

Comment:
Can you please make this throw a 500 if there are multiple PKs rather? Otherwise we just get "weird" behavior.

Reply:
added a check.